### DISCUSSION TITLE
Could Detours hook WinRT API ?

### DISCUSSION THREAD

**Discussion by marcus1877** [microsoft/Detours#239] [Q&A] [answered]
Hi ,
I'm a QA testing our UWP APP , which is leveraging WinRT to do something . Could Detours hook WinRT APIs ? 
If it does , we can do some awesome tests .
Tks.

**Comment by sylveon** [comment, marked as answer]
Sure. WinRT APIs are at their core COM interfaces. You can hook them [like you can hook other COM APIs](https://github.com/microsoft/Detours/tree/master/samples/commem). You'll most likely want to hook RoGetActivationFactory and RoActivateInstance, then proceed from there.

However, instead of looking at detours to test an app, I would suggest abstracting away the WinRT API from the product code so that you can write a mock for testing.

**Reply by marcus1877** [reply]
Cool !
